servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/DefaultExecutor.java

```diff
@@ -157,7 +158,7 @@ private static void shutdownExecutor(java.util.concurrent.Executor jdkExecutor)
             try {
                 ((AutoCloseable) jdkExecutor).close();
             } catch (Exception e) {
-                throw new RuntimeException("unexpected exception while closing executor: " + jdkExecutor, e);
+                throwException(e);
```

Comment:
`jdkExecutor` is an `AutoCloseable` which is subject to those rules, we are just re-throwing w/e it throws. We aren't implementing `AutoCloseable` here, do we need to do anything different?

Reply:
This method is invoked from `AutoCloseable#close()` method we implement in line 197. I believe javadoc means a try-with-resources use-case. If the thread is interrupted when the `close()` is invoked, the `InterruptedException` will be added as suppressed exception to the originally thrown exception. 
We can say it's a bug of original `jdkExecutor` bcz we just re-throwing or we can try to handle that bug in a better way. Either way works for me. If you keep re-throw, please add a comment saying that we do not expect `InterruptedException` here and the `jdkExecutor` should guarantee that.

Reply:
> If the thread is interrupted when the close() is invoked, the InterruptedException will be added as suppressed exception to the originally thrown exception.

Can you clarify when `InterruptedException` would be added as a suppressed as some other exception? This seems like a bug (unless the interrupt flag is set on the Thread).

Reply:
```
try (AutoCloseable res = newResource()) {
    throw new Exception("foo");
} catch (Exception e) {
    // if `AutoCloseable#close()` throws `InterruptedException`, it will be added as suppressed exception of Exception("foo")
}
```
more details here: https://stackabuse.com/the-try-with-resources-statement-in-java/

Reply:
gotcha thx. Doesn't seem like it should be the caller's responsibility to try to correct as there is some ambiguity in desired behavior depending upon the usage:

```
catch (InterruptedException e) {
  // this may "spuriously" interrupt again when the next interruptible operation
  // occurs, which is likely not desirable.
  Thread.currentThread().interrupt(); 
  // throw something...
  // if we re-throw InterruptedException we have already set the interrupt state which may lead to
  //   more undesirable/forced interrupts
  // if we throw something else (maybe `CompletionException`) then caller has to unwrap it to know
  //   it was bcz of an interrupt
}

Reply:
ok, let's keep the rethrow